### PR TITLE
feat(chart)!: use global redis

### PR DESCRIPTION
**Pull request by olevski**
Hopefully I did this right. It is really cool that you use typescript in the ui server. I did not know this.

With this change everything the ui server needs is specified in the global redis section. So if someone brings their own redis then they do not install the redis chart bundled with renku but they fill out all stuff in global.redis. If people use the included redis then they can stick with the defaults and everything should work. If they change the defaults then the global.redis section and regular redis section need to be compatible and some values between the two will be duplicated.

**THIS IS A BREAKING CHANGE FOR THE HELM CHART**

/deploy #persist

~Please add any comments on the helm chart and its organization in the parent PR in the renku repo at https://github.com/SwissDataScienceCenter/renku/pull/2483. That way we can all agree on the any potential changes to the helm chart.~

### CODE REVIEW

**Comment by lorenzo-cavazzi** [comment]
Hey, not sure if it's relevant but there was already a stub on using the global redis instance on the UI server #1554

**Comment by olevski** [comment]
> Hey, not sure if it's relevant but there was already a stub on using the global redis instance on the UI server #1554

I am sorry Lorenzo, I missed that. I should have looked first. I will close this PR and add the changes on the original one.

**Comment by lorenzo-cavazzi** [comment]
Don't worry, the other PR is now outdated (looking at "File changed" is very confusing), and it's probably better to keep the structure in line with the other repos.
When this is ready, I can quickly check if there is anything useful in the other PR that I can port here, and then close it.

**Comment by olevski** [comment]
@lorenzo-cavazzi this is ready for review and testing. All other dependent/required PRs have been merged.

File: server/src/config.ts [hunk 1]
```diff
@@ -63,6 +63,8 @@ const REDIS = {
   port: convertType(process.env.REDIS_PORT) || 6379,
   database: process.env.REDIS_DATABASE || "0",
   password: process.env.REDIS_PASSWORD || null,
+  isSentinel: (process.env.REDIS_IS_SENTINEL || "false") == "true",
```
Comment (lorenzo-cavazzi) [inline server/src/config.ts:66]:
```suggestion
  isSentinel: convertType(process.env.REDIS_IS_SENTINEL) || false,
```

File: server/src/storage/RedisStorage.ts [hunk 2]
```diff
@@ -27,19 +27,31 @@ class RedisStorage implements Storage {
...
-      host,
-      port,
       lazyConnect: true,
       retryStrategy: (times) => {
         return times > 5 ?
           null :
           10000;
       },
     };
+    if (isSentinel){
```
Comment (lorenzo-cavazzi) [inline server/src/storage/RedisStorage.ts:43]:
```suggestion
    if (isSentinel) {
```

File: server/src/storage/RedisStorage.ts [hunk 3]
```diff
@@ -27,19 +27,31 @@ class RedisStorage implements Storage {
...
+      redisConfig.sentinels = [
+        { host, port },
+      ];
+      redisConfig.name = masterSet;
+      if (password) {
```
Comment (lorenzo-cavazzi) [inline server/src/storage/RedisStorage.ts:48]:
```suggestion
      if (password)
```

File: server/src/storage/RedisStorage.ts [hunk 4]
```diff
@@ -27,19 +27,31 @@ class RedisStorage implements Storage {
...
+        redisConfig.sentinelPassword = password;
+      }
```
Comment (lorenzo-cavazzi) [inline server/src/storage/RedisStorage.ts:50]:
```suggestion
```

File: server/src/storage/RedisStorage.ts [hunk 5]
```diff
@@ -27,19 +27,31 @@ class RedisStorage implements Storage {
...
+    } else {
```
Comment (lorenzo-cavazzi) [inline server/src/storage/RedisStorage.ts:51]:
```suggestion
    }
    else {
```

File: server/src/storage/RedisStorage.ts [hunk 6]
```diff
@@ -27,19 +27,31 @@ class RedisStorage implements Storage {
   constructor(
     host: string = config.redis.host,
     port: number = config.redis.port as number,
-    password: string = config.redis.password
+    password: string = config.redis.password,
+    isSentinel: boolean = config.redis.isSentinel,
```
Comment (lorenzo-cavazzi) [inline server/src/storage/RedisStorage.ts:31]:
```suggestion
    isSentinel: boolean = config.redis.isSentinel as boolean,
```

**Review by lorenzo-cavazzi — CHANGES_REQUESTED** [review]
Hey Tasko, sorry for the delay!
There are a few adjustments to the code, mainly to comply with the lining rules.

You can add them into a commit and re-run the acceptance tests to see if it works -- mind that I changed the deploy string to include some requirements for the current UI master branch to work with tests

**Review by lorenzo-cavazzi — DISMISSED** [review]
Thank you Tasko for this PR! It works well :+1:

**Review by lorenzo-cavazzi — APPROVED** [review]

**Pull request merged**